src/pyupgrade/plugins/unnecessary_typed_dict_syntactic_form.rs

```diff
@@ -0,0 +1,194 @@
+            )),
+            value: None,
+            simple: 1,
+        },
+    )
+}
+
+fn create_pass_stmt() -> Stmt {
+    Stmt::new(Default::default(), Default::default(), StmtKind::Pass)
+}
+
+fn create_classdef_stmt(class_name: &String, body: Vec<Stmt>, total: Option<KeywordData>) -> Stmt {
+    let keywords = match total {
+        Some(keyword) => vec![Keyword::new(
+            Default::default(),
+            Default::default(),
+            keyword,
+        )],
+        None => vec![],
+    };
+    Stmt::new(
+        Default::default(),
+        Default::default(),
+        StmtKind::ClassDef {
+            name: class_name.to_string(),
+            bases: vec![Expr::new(
+                Default::default(),
+                Default::default(),
+                ExprKind::Name {
+                    id: "TypedDict".to_string(),
+                    ctx: ExprContext::Load,
+                },
+            )],
+            keywords,
+            body,
+            decorator_list: vec![],
+        },
+    )
+}
+
+fn get_properties_from_dict_literal(keys: &[Expr], values: &[Expr]) -> Result<Vec<Stmt>> {
+    keys.iter()
+        .zip(values.iter())
+        .map(|(key, value)| match &key.node {
+            ExprKind::Constant {
+                value: Constant::Str(property),
+                ..
+            } => Ok(create_property_assignment_stmt(property, &value.node)),
+            _ => bail!("key is not a Str"),
+        })
+        .collect()
+}
+
+fn get_properties_from_dict_call(func: &Expr, keywords: &[Keyword]) -> Result<Vec<Stmt>> {
+    match &func.node {
+        ExprKind::Name { id: func_name, .. } => match func_name.as_str() {
+            "dict" => keywords
+                .iter()
+                .map(|keyword| {
+                    let property = &keyword.node.arg.clone().ok_or_else(|| anyhow!("no arg"))?;
+                    Ok(create_property_assignment_stmt(
+                        property,
+                        &keyword.node.value.node,
+                    ))
+                })
+                .collect(),
+            _ => bail!("func is not dict"),
+        },
+        _ => bail!("func is not a Name"),
+    }
+}
+
+// Deprecated in Python 3.11, Removed in Python 3.13?
+fn get_properties_from_keywords(keywords: &[Keyword]) -> Result<Vec<Stmt>> {
+    keywords
+        .iter()
+        .map(|keyword| {
+            let property = &keyword.node.arg.clone().ok_or_else(|| anyhow!("no arg"))?;
+            Ok(create_property_assignment_stmt(
+                property,
+                &keyword.node.value.node,
+            ))
+        })
+        .collect()
+}
+
+fn try_to_fix(
+    check: &mut Check,
+    stmt: &Stmt,
+    class_name: &String,
+    args: &[Expr],
+    keywords: &Vec<Keyword>,
+) -> Result<()> {
+    let mut total: Option<KeywordData> = None;
+    let body = if let Some(dict) = args.get(1) {
+        total = keywords.get(0).and_then(|keyword| match &keyword.node.arg {
```

Comment:
Can we pull this assignment into a separate series of `if` statements, or return a tuple from these branches? I think it'd be a bit clearer than using this branch to assign to `total` as a side-effect.